### PR TITLE
Equivariant test

### PR DESCRIPTION
**Pull request by yuanqing-wang**
## Description
Add equivariance / invariance tests to all models.

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [x] Add rotation / translation / reflection random functions 
- [x] Apply random functions on input and test consistency

## Questions
- [ ] Question1

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by yuanqing-wang** [comment]
@wiederm @ArnNag could you take a look at the PaiNN implementation and see what's going on resulting in the failures?

**Comment by wiederm** [comment]
This PR is ready to merge --- the test failures are due to the `gdown` package, all other tests are now passing locally.

**Comment by chrisiacovella** [comment]
> This PR is ready to merge --- the test failures are due to the `gdown` package, all other tests are now passing locally.

I'm not able to to get these tests to run locally (gdown does not seem to be working properly; I've tried different versions and using the "fuzzy" approach, but still end up with errors).  I've been trying to isolate the exact problem (tests outside of model forge seem to work).

**Review by chrisiacovella — APPROVED** [review]
I think the only change would be to better annotate the cutoff. (maybe change to 0.5 and note that it should be in nanometers).

File: devtools/conda-envs/test_env.yaml [hunk 1]
```diff
@@ -14,13 +14,12 @@ dependencies:
   - qcportal>=0.50
   - pytorch>=2.1
   - loguru
-  - gdown
+  - gdown>=5.0.0
```
[the remark this answers is not in the document]
Comment (wiederm) [reply devtools/conda-envs/test_env.yaml:17]: Agreed. we need to find a better way to handle this. Converted the comment to an issue.

File: modelforge/potential/models.py [hunk 2]
```diff
@@ -24,22 +26,25 @@ class PairList(nn.Module):
         Forward pass for PairList.
     """
 
-    def __init__(self, cutoff: float = 5.0):
+    def __init__(self, cutoff: float = 5.0, only_unique_pairs: bool = False):
```
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/potential/models.py:29]: This requires some refactoring. I will do this in a new PR.

File: modelforge/potential/utils.py [hunk 3]
```diff
@@ -373,8 +409,23 @@ def neighbor_pairs_nopbc(
...
+        i_indices, j_indices = torch.meshgrid(
+            torch.arange(start=0, end=n, dtype=torch.int64),
+            torch.arange(start=0, end=n, dtype=torch.int64),
+        )
+        # remove indices for which i_indices == j_indices
+        mask = i_indices != j_indices
+        i_indices = i_indices[mask]
+        j_indices = j_indices[mask]
+
+        # i_indices = i_indices.reshape(-1)
```
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/potential/utils.py:456]: done!

**Pull request merged**